processor/src/decoder/aux_trace/op_group_table.rs

```diff
@@ -88,18 +91,23 @@ fn get_op_group_table_removal_multiplicand<E: FieldElement<BaseField = Felt>>(
 ) -> E {
     let group_count = main_trace.group_count(i);
     let block_id = main_trace.addr(i);
+    let group_value = {
+        let op_code = main_trace.get_op_code(i);
 
-    let op_code = main_trace.get_op_code(i);
-    let tmp = if op_code == Felt::from(OPCODE_PUSH) {
-        main_trace.stack_element(0, i + 1)
-    } else {
-        let h0 = main_trace.decoder_hasher_state_first_half(i + 1)[0];
+        if op_code == Felt::from(OPCODE_PUSH) {
+            main_trace.stack_element(0, i + 1)
+        } else if op_code == Felt::from(OPCODE_EMIT) {
+            main_trace.decoder_hasher_state(i)[EMIT_IMM_VALUE_HASHER_TRACE_IDX]
```

Comment:
I would probably use `main_trace.helper_register()` here. The main reason is to logically separate helper registers from hasher state as in the future helper registers may arranged differently.

Reply:
fixed. Also removed `EMIT_IMM_VALUE_HASHER_TRACE_IDX`